commands/host.vegman

```diff
@@ -193,35 +238,64 @@ function cmd_virtualmedia_mount {
     return 1
   fi
 
-  if [[ ! -e ${imagefile} ]] || [[ -d ${imagefile} ]]; then
-    echo "Please specify the path to image file" >&2
+  if ! [[ ${imagefile} =~ ${imagefile_pattern} ]]; then
+    echo "Please specify the valid path to the image file" >&2
     return 1
   fi
 
-  usb-ctrl insert "usb.ms.cli" "${imagefile}" "${type}"
+  if [[ ! -z ${username} ]] || [[ ! -z ${password} ]]
+  then
+    printf %b "${username:0:64}\0${password:0:64}\0" >6
```

Comment:
@nest1ing, I very worry about packing and secure passing credentials via PIPE.
The virtual media expect blob `%{username}\0%{password}`, but is there possible a case when `printf  %b` might cut blob by first `\0`?

Reply:
`%b` is a bash-specific modifier. I assume it will work as you expect and will not truncate the result string.
But if you doubt replace it with the standard C-style printf format:
```bash
printf "%s\0%s\0" "${username:0:64}" "${password:0:64}" >6
```

Reply:
There is a thing makes me confused a little bit more than `printf` format string:
What is `6`? It looks like a file name, but not a PIPE.
Who should receive this data?

Reply:
Yes, it looks like a mistake.
The `FD` of PIPE should be passed to the DBus `Mount` method, and the `virtual media` service will accept the PIPE and read credentials via it.
I guess I should open a new pipe first as FIFO, associate it to some tmpfile, rm file, and after correct read, do unlink FD.

Reply:
fixed